plugins/modules/riak.py

```diff
@@ -126,7 +126,7 @@ def main():
 
     # make sure riak commands are on the path
     riak_bin = module.get_bin_path('riak')
-    riak_admin_bin = module.get_bin_path('riak-admin')
+    riak_admin_bin = module.get_bin_path('riak-admin') or riak_bin + ' admin'
```

Comment:
Let's not go back to string manipulation for executables. Please use a list:
```suggestion
    riak_admin_bin = module.get_bin_path('riak-admin')
    riak_admin_bin = [riak_admin_bin] if riak_admin_bin is not None else [riak_bin, ' admin']
```
This makes some more changes necessary, though, since `riak_admin_bin` is used in more cases.

Alternatively have two versions of it, one a list and one a string (maybe `riak_admin_bin` and `riak_admin_bin_list`?), so depending on the context one or the other can be used.